pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -117,19 +122,56 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 	}
 
 	status := history.Phase
-	reqLogger.Info("current cluster status", "status", status)
+
+	// status should never be an empty string at this stage. Explicity test and return error if it is.
+	if status == "" {
+		return reconcile.Result{}, fmt.Errorf("Status should never be undefined at this stage")
+	}
+
+	reqLogger.Info("Current cluster status", "status", status)
 
 	switch status {
-	case "", upgradev1alpha1.UpgradePhaseNew, upgradev1alpha1.UpgradePhasePending:
-		// TODO verify if it's time to do upgrade, if no, set to "pending", if it's yes, perform upgrade, and set status to "upgrading"
-		reqLogger.Info("checking whether it's ready to do upgrade")
+	case upgradev1alpha1.UpgradePhaseNew, upgradev1alpha1.UpgradePhasePending:
+		reqLogger.Info("Validating UpgradeConfig")
+
+		// Get current ClusterVersion
+		cv, err := osd_cluster_upgrader.GetClusterVersion(r.client)
```

Comment:
A minor nit that doesn't necessarily need addressing here, but maybe we eventually need to refactor the code used by GetClusterVersion (and the GetCurrentVersion used by the validator) into something outside of the OSD Cluster Upgrader. Mainly because the controller and the validator are pretty OCP-agnostic things that now have a dependency on code sitting in the OSD-specific package, which gives me a few irritations design-wise.

Reply:
+1 can we add to https://issues.redhat.com/browse/OSD-4168?

Reply:
added